src/MessageRingBuffer.cpp

```diff
@@ -68,18 +80,18 @@ void MessageRingBuffer::push(ShapeShifterMsg::ConstPtr msg, const ros::Time& rec
         {
             // we accumulate all dropped messages to invoke the droppedCallback after unlocking.
             // The callback might take a long time and potentially lock other resources causing a deadlock
-            currentSize -= buffer.front().msg->objectSize() + sizeof(ros::Time);
+            currentSize -= getMemorySize(buffer.front());
             droppedMsgs.emplace_back(std::move(buffer.front()));
             buffer.pop_front();
         }
         if (currentSize + entrySize <= maxSize)
         {
             currentSize += entrySize;
-            buffer.emplace_back(std::move(msg), receiveTime);
+            buffer.push_back(std::move(e));
```

Comment:
```suggestion
            buffer.emplace_back(std::move(e));
```

Reply:
push_back with a move is faster than emplace with move